nixos/modules/services/networking/networkmanager.nix

```diff
@@ -140,6 +144,34 @@ in {
         '';
       };
 
+      ethernet.macAddress = mkOption {
+        type = types.nullOr (types.either types.str (types.enum ["permanent" "preserve" "random" "stable"]));
```

Comment:
I did this so that the networkmanager could choose the default value (if none has been set).

But after checking the blog-post, it might be better to use `"preserve"`

> Update-2017-01-25: with 1.6 release and newer, the default value changed from “permanent” to “preserve” [[commit](https://cgit.freedesktop.org/NetworkManager/NetworkManager/commit/?id=fae5ecec5a4d9987a1915441602cb78275a9f490)],[[bug](https://bugzilla.gnome.org/show_bug.cgi?id=770611)].

By setting the default to `"preserve"` might acctually fix the issue with setting `networking.interfaces.wlan0.macAddress` while the networkmanger is used.

I will change the default to `"preserve"`